libarchive/archive_read_support_format_rar5.c

```diff
@@ -2410,6 +2425,14 @@ static int rar5_read_header(struct archive_read *a,
 	struct rar5* rar = get_context(a);
 	int ret;
 
+	/*
+	 * It should be sufficient to call archive_read_next_header() for
+	 * a reader to determine if an entry is encrypted or not.
+	 */
+	if (rar->has_encrypted_entries == ARCHIVE_READ_FORMAT_ENCRYPTION_DONT_KNOW) {
+		rar->has_encrypted_entries = 0;
+	}
```

Comment:
This is something I didn't necessarily like doing, but I copied from other places. Encryption might not necessarily be detected when reading the first header (e.g. if the Nth entry is the first encrypted one), however there's not much difference in practice between perpetually returning `ARCHIVE_READ_FORMAT_ENCRYPTION_DONT_KNOW` and returning `0`.